src/re2/gen_build_yaml.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python2.7
+
+# Copyright 2015 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import os
+import sys
+import glob
+import yaml
+
+os.chdir(os.path.dirname(sys.argv[0]) + '/../..')
+
+out = {}
+
+out['libs'] = [{
+    'name':
```

Comment:
adding the `re2` library this way works, but depending on how the corresponding bazel BUILD file works, you might be able to get away with just extracting the list of source files from bazel BUILD automatically.

I think for the first version of the PR this is optional, but I wanted to mentions that there might be a better and cleaner way of doing this and it's something you should look into after this PR is in.

Reply:
added todo

Reply:
nit: replace "for optimization" / "to reduce duplication".

Reply:
done